contracts/src/peer_protocol.cairo

```diff
@@ -413,12 +442,35 @@ mod PeerProtocol {
 
             self.proposals.entry(proposal_id).is_accepted.write(true);
 
-            self.emit(ProposalAccepted {
-                proposal_type: proposal.proposal_type,
-                accepted_by: caller,
-                token: proposal.token,
-                amount: proposal.amount
-            });
+            self
+                .emit(
+                    ProposalAccepted {
+                        proposal_type: proposal.proposal_type,
+                        accepted_by: caller,
+                        token: proposal.token,
+                        amount: proposal.amount
+                    }
+                );
+        }
+
+        fn repay_proposal(ref self: ContractState, proposal_id: u256) {
```

Comment:
This only records the details. Just like with the Lender acceptance function or the borrower acceptance function, there has to be a transaction to actually repay the owed debt and the interest on it.